### DISCUSSION TITLE
Is it safe to share webpack cache?

### DISCUSSION THREAD

**Discussion by gajus** [webpack/webpack#15474] [Q&A]
Referring to this configuration:

https://webpack.js.org/configuration/cache/

Would it be safe to set it to:

```js
cache: {
  cacheDirectory: '/tmp/.webpack_cache',
  type: 'filesystem',
},
```

The idea is that we would use this configuration on a CI/CD runner VM. However, the catch is that there are other webpack projects built on the same VM.

The goal is to speed up webpack builds on VMs that run tests.

**Comment by vankop** [comment]
you can share cache between webpack builds in CI/CD. moving local cache to CI/CD does not work (at least for now)

**Reply by gajus** [reply]
But, if there are multiple projects sharing the same webpack cache directory, how does webpack know if this is the correct cache to use?

```
contra-web-app on main [$!⇣] on ☁️  gajus@contra.com
at 09:27:33 ❯ ls -lah .webpack_cache/
total 0
drwxr-xr-x   3 gajus  staff    96B Mar  4 09:14 .
drwxr-xr-x@ 63 gajus  staff   2.0K Mar  4 09:14 ..
drwxr-xr-x   4 gajus  staff   128B Mar  4 09:14 default-development
contra-web-app on main [$!⇣] on ☁️  gajus@contra.com
at 09:27:36 ❯ ls -lah .webpack_cache/default-development
total 4985688
drwxr-xr-x  4 gajus  staff   128B Mar  4 09:14 .
drwxr-xr-x  3 gajus  staff    96B Mar  4 09:14 ..
-rw-r--r--  1 gajus  staff   2.4G Mar  4 09:14 0.pack
-rw-r--r--  1 gajus  staff    11M Mar  4 09:14 index.pack
```

The files themselves have no indication of hash or another mechanism to distinguish projects.

**Reply by vankop** [reply]
multiple projects should have different cache directory.

**Reply by gajus** [reply]
What happens if cache were to be shared by different projects?, i.e. is there a risk of corrupted build, or would it just ignore cache?

**Reply by vankop** [reply]
> there a risk of corrupted build

technically, no.

**Comment by Djiit** [comment]
Hey @vankop, I'm exploring ways of sharing/reusing caches from the CI on local devs machines. Is it still out of the radar?

**Reply by vankop** [reply]
yes, it is not possible
